File: clients/python_pypy/my_strategy.py
from model.game import Game
from model.order import Order
from model.constants import Constants
from typing import Optional
from debug_interface import DebugInterface


class MyStrategy:
    def __init__(self, constants: Constants):
        pass
    def get_order(self, game: Game, debug_interface: Optional[DebugInterface]) -> Order:
        return Order({})
    def debug_update(self, debug_interface: DebugInterface):
        pass
    def finish(self):
        pass

File: quickstart/MyStrategy.java
package ai_cup_22;

import ai_cup_22.model.*;

public class MyStrategy {
    public MyStrategy(Constants constants) {}
    public Order getOrder(Game game, DebugInterface debugInterface) {
        java.util.HashMap<Integer, UnitOrder> orders = new java.util.HashMap<>();
        for (Unit unit : game.getUnits()) {
            if (unit.getPlayerId() != game.getMyId()) {
                continue;
            }
            orders.put(unit.getId(), new UnitOrder(
                new Vec2(-unit.getPosition().getX(), -unit.getPosition().getY()),
                new Vec2(-unit.getDirection().getY(), unit.getDirection().getX()),
                new ActionOrder.Aim(true)));
        }
        return new Order(orders);
    }
    public void debugUpdate(DebugInterface debugInterface) {}
    public void finish() {}
}